### DISCUSSION TITLE
Emulated keyboard input doesn't work

### DISCUSSION THREAD

**Discussion by SonoSooS** [ppy/osu#18583] [Ideas]
### Type

Game behaviour

### Bug description

I have a few DIY wireless tablets, all of which use the same feeder application for input.

The problem is, that while mouse emulation works as exepected, all of the keyboard input emulation is not picked up by lazer.

I implemented the feeder application in a way, so that it works on Windows, Linux, and macOS as well, but sadly that means that raw input has to be disabled, as it doesn't register itself as a HID input device.

osu!stable does pick up [keybd_event()](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-keybd_event) as expected, but lazer doesn't.

### Version

2022.515.0, 2022.615.1

### Logs

[runtime.log](https://github.com/ppy/osu/files/8843025/runtime.log)
[network.log](https://github.com/ppy/osu/files/8843026/network.log)
[database.log](https://github.com/ppy/osu/files/8843027/database.log)
[performance.log](https://github.com/ppy/osu/files/8843028/performance.log)
[updater.log](https://github.com/ppy/osu/files/8843029/updater.log)

**Comment by peppy** [comment]
Keeping this as a feature request. We haven't had a single user run into this and wouldn't consider adding support unless it's a common issue.

**Comment by AuCrossover** [comment]
I also have this problem when using Android 12. I tried 2 different keyboards and attempted to use osu!lazer. Both worked when using the browser, but not in lazer. Only the mouse was usable and it felt like I was playing the web version.

**Reply by peppy** [reply]
That is not going to be the same issue. Please search for android specific input issues as I believe this has already been reported.
